### PR TITLE
The Drawer should animate out when popped off the navigator

### PR DESCRIPTION
**Pull request by abarth**

### CODE REVIEW

File: sky/packages/sky/lib/src/widgets/navigator.dart [hunk 1]
```diff
@@ -177,7 +177,7 @@ abstract class Route {
     _performance = createPerformance();
   }
 
-  PerformanceView get performance => _performance?.view;
```
Comment (Hixie) [inline sky/packages/sky/lib/src/widgets/navigator.dart]: This is too powerful.

As per our discussion, I think PerformanceView should expose an "obtainWriteHandle" method (or whatever) that gives you a handle to modify the performance, but which can be revoked. When the current route is not this route or the route below it (the stock home, e.g.) then you can't obtain a handle. When navigator pops or pushes, you lose your handle. When another handle is obtain, you lose your handle.

**Comment by abarth** [comment]
@Hixie Ready for another look.

File: sky/packages/sky/lib/src/widgets/drawer.dart [hunk 2]
```diff
@@ -34,120 +34,114 @@ const Duration _kThemeChangeDuration = const Duration(milliseconds: 200);
...
+
+  void didPush(NavigatorState navigator) {
+    super.didPush(navigator);
+    _performance.forward();
+  }
+
+  void didPop([dynamic result]) {
+    super.didPop(result);
+    _performance.reverse();
+    _performance = null;
```
Comment (Hixie) [inline sky/packages/sky/lib/src/widgets/drawer.dart]: why do you set it to null here?

**Comment by Hixie** [comment]
LGTM, but file a bug about not supporting catching the drawer (and repushing the drawer route) while it's closing.

File: sky/packages/sky/lib/src/widgets/drawer.dart (hunk 2 above)
Comment (abarth) [inline sky/packages/sky/lib/src/widgets/drawer.dart]: That's so the gestures can't modify the performance anymore after this point.

**Comment by abarth** [comment]
Done: https://github.com/flutter/engine/issues/1539

**Pull request merged**